application/backend/src/main/resources/apps/typerefinery/components/structure/page/clientlibs/functions.js

```diff
@@ -64,15 +67,9 @@ window.MessageService.Client = MessageService.Client || {};
                         if ($component) {
                             if ($component.getAttribute('data-module') === 'tickerComponent') {
                                 componentNs?.Widgets?.Ticker?.dataReceived(payload.data, $component)
-                            }else if ($component.getAttribute('data-module') === 'linechartComponent') {
-                                componentNs?.Graphs?.LineChart?.dataReceived(payload.data, $component)
-                            }else if ($component.getAttribute('data-module') === 'piechartComponent') {
-                                componentNs?.Graphs?.PieChart?.dataReceived(payload.data, $component)
-                            }else if ($component.getAttribute('data-module') === 'xaxisbarchartComponent') {
-                                componentNs?.Graphs?.XAxisBarChart?.dataReceived(payload.data, $component)
-                            }else if ($component.getAttribute('data-module') === 'barchartComponent') {
-                                componentNs?.Graphs?.BarChart?.dataReceived(payload.data, $component);
-                            }else if ($component.getAttribute('data-module') === 'tableComponent') {
+                            } else if ($component.getAttribute('data-module') === 'chartComponent') {
```

Comment:
each one of these should be in its own behaviour.js!

Reply:
we cant be updating some crazy monolith if block...

Reply:
Then whether the event listners should also be in the behaviour.js ?

![image](https://user-images.githubusercontent.com/54505967/217431143-512560ea-aa16-4604-8d1b-f6a02b4fd830.png)